src/CaptureClient/include/CaptureClient/ApiEventIdSetter.h

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef CAPTURE_CLIENT_API_EVENT_ID_SETTER_H_
+#define CAPTURE_CLIENT_API_EVENT_ID_SETTER_H_
+
+#include <cstdint>
+
+#include "ClientData/TimerTrackDataIdManager.h"
+#include "GrpcProtos/capture.pb.h"
+
+namespace orbit_capture_client {
+
+class ApiEventIdSetter {
+ public:
+  virtual ~ApiEventIdSetter() = default;
+
+  virtual void SetId(TimerInfo& timer_info);
+
+ protected:
+  virtual uint64_t GetId(const TimerInfo& timer_info) = 0;
+};
+
+// Sets equal `api_scope_group_id` to the instances of TimeInfo
+// if and only if their `api_scope_name` are equal.
+// The ids are chosen consequtively starting with `start_id`.
+//
+// To instantiate, use `NameEqualityApiEventIdSetter::Create`
+// as this ensures no overlap between `api_scope_group_id`
+// and `function_id`.
+class NameEqualityApiEventIdSetter : public ApiEventIdSetter {
```

Comment:
Not sure if we already want to split this into two files.

Reply:
Well, after I have removed the `SetId`, the file is shorter. Do you still wish to split it?